include/server/mir/scene/surface.h

```diff
@@ -142,6 +142,16 @@ class Surface :
 
     /// The session this surface was created by
     virtual auto session() const -> std::weak_ptr<Session> = 0;
+
+    /// Sets the dimensions of the window frame
+    /// It is assumed that any decorations are provided by a child window
+    /// Frame size is not directly accessable, but can be determined by comparing client_size() and frame_size()
```

Comment:
What is `frame_size()`?

If it's a misnomer for `window_size()` then maybe this function should be `set_window_margins()`?